pkg/controller/csi_handler.go

```diff
@@ -114,8 +215,8 @@ func (h *csiHandler) SyncNewOrUpdatedVolumeAttachment(va *storage.VolumeAttachme
 }
 
 func (h *csiHandler) syncAttach(va *storage.VolumeAttachment) error {
-	if va.Status.Attached {
-		// Volume is attached, there is nothing to be done.
+	if !h.consumeForceSync(va.Name) && va.Status.Attached {
```

Comment:
Can we be sure that the va we pulled from the queue is the one we pushed in `ReconcileVA`? What if some other operation is sitting near the front of the queue when we reconcile? Hopefully that's fine?

- if the two queued vas agree on state, then the second one will be skipped but that's fine
- if the two queued vas disagree on state, then they will both pass the already attached/detached check?

Reply:
the `forceSync` flag is based on `vaName` so will not affect any other items on the queue. When the queue eventually pops the item we have requeued for force sync it will go into this modified logic